src/passes/NoInline.cpp

```diff
@@ -48,8 +48,10 @@ struct NoInline : public Pass {
   NoInline(NoInlineMode mode) : mode(mode) {}
 
   void run(Module* module) override {
-    std::string pattern = getPassOptions().getArgument(
-      name, "Usage usage:  wasm-opt --" + name + "=WILDCARD");
+    std::string pattern =
+      passArg ? *passArg
+              : getPassOptions().getArgument(
```

Comment:
I wonder if we can simplify this? Specifically that it accepts two mechanisms seems a little more complex than it would ideally be.

In fact `--passname=ARG` sets both of them, so I think we only need to check the former? But I might be missing something.

Reply:
The reason why I kept the global mechanism is `--pass-arg`.

Reply:
Ah, yes, I agree we want the global mechanism too. And that makes sense: there can be a global flag that multiple passes (even of different operations, not just instances of the same Pass) care about, but also each pass instance can have it's own. But here, I think this can always read `passArg` rather than check `passArg` and maybe read `getPassOptions().getArgument(..)`.